src/java/com/puppetlabs/http/client/impl/PersistentSyncHttpClient.java

```diff
@@ -26,10 +31,42 @@ private static void logAndRethrow(String msg, Throwable t) {
         throw new HttpClientException(msg, t);
     }
 
+    public Map<String, Timer> getClientMetrics(){
+        return Metrics.getClientMetrics(metricRegistry);
+    }
+
+    public Map<String, Timer> getClientMetrics(String url, Metrics.MetricType metricType) {
```

Comment:
I wonder if it's worth it to repeat these `get*Metrics` helper methods across both of the HTTP client classes or just have clients interact with the underlying `Metrics` class directly.  I'd probably opt for the latter - better separation of concerns, I think.  If we do want these methods on both client classes, though, I'd be inclined to create an abstract base class which implements them and then have the two Persistent*HttpClient classes extend from that base class - so that we don't have to duplicate the implementation across the classes.